lib/modules/datasource/hermit/index.ts

```diff
@@ -0,0 +1,153 @@
+      `${registryUrl}-${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    logger.trace(`HermitDataSource.getReleases()`);
+
+    if (!registryUrl) {
+      throw new Error('registryUrl must be supplied');
+    }
+
+    const res = await this.getHermitRelease(packageName, registryUrl);
+    const sourceUrl = res.Repository;
+
+    return {
+      sourceUrl,
+      releases: [
+        ...res.Versions.map((v) => ({
+          version: v,
+          sourceUrl,
+        })),
+        ...res.Channels.map((v) => ({
+          version: v,
+          sourceUrl,
+        })),
+      ],
+    };
+  }
+
+  /**
+   * getHermitSearchManifest fetch the index.json from release
+   * named index, parses it and returned the parsed JSON result
+   */
+  @cache({
+    namespace: `datasource-${datasource}-search-manifest`,
+    key: ({ registryUrl }: GetReleasesConfig) => registryUrl ?? '',
+  })
+  async getHermitSearchManifest(
+    registryUrl: string
+  ): Promise<HermitSearchResult[]> {
+    if (!registryUrl.startsWith('https://github.com')) {
+      throw new Error(`Only Github registryUrl is supported`);
+    }
+
+    const u = new URL(registryUrl);
+    const host = u.host;
+    const path = u.pathname;
+
+    const parts = path.split('/').filter((p) => p !== '');
+
+    if (parts.length < 2) {
+      throw new Error(`can't find owner & repo in the url`);
+    }
+
+    logger.debug(
+      { registryUrl },
+      `fetching hermit search manifest from repository's index release`
+    );
+
+    const owner = parts[0];
+    const repo = parts[1];
+    const apiBaseUrl = getApiBaseUrl(`https://${host}`);
+
+    const indexRelease = await this.githubHttp.getJson<GithubRelease>(
+      `${apiBaseUrl}repos/${owner}/${repo}/releases/tags/index`
+    );
+
+    const asset = indexRelease.body.assets.find(
+      (asset: GithubReleaseAsset) => asset.name === 'index.json'
+    );
+
+    if (!asset) {
+      throw new Error(
+        `cannot find asset index.json in the given registryUrl ${registryUrl}`
+      );
+    }
+
+    // fetches the content of the asset via calling back to the url
+    // (not browser_download_url) to work with private
+    // repositories
+    const indexContent = await this.githubHttp.get(asset.url, {
+      headers: {
+        accept: 'application/octet-stream',
+      },
+    });
+
+    return JSON.parse(indexContent.body) as HermitSearchResult[];
+  }
+
+  /**
+   * getHermitRelease fetches the search result manifest and search package name inside
+   */
+  async getHermitRelease(
+    name: string,
+    registryUrl: string
+  ): Promise<HermitSearchResult> {
+    const items = await this.getHermitSearchManifest(registryUrl);
+
+    const searchResult = items.find((i) => i.Name === name);
+
+    if (!searchResult) {
+      throw new Error(`cannot find package ${name} in the search result`);
```

Comment:
log a debug message and return null. we prefer to not throw errors from datasource